net: zperf: Add periodic reporting for upload

nxpadamm opened the pull request:
Add option (-i) for zperf tcp upload that will enable periodic result reporting. This is useful for monitoring performance swings during a longer session.

Example zperf shell usage:
```
uart:~$ zperf tcp upload -a -i 2 192.0.2.2 5001 10
Remote port is 5001
Connecting to 192.0.2.2
Duration:       10.00 s
Packet size:    256 bytes
Rate:           10 kbps
Starting...
Duration: 2.00 s | Packets:  11051 | Errors:      0 | Rate: 11.31 Mbps
Duration: 2.00 s | Packets:  11072 | Errors:      0 | Rate: 11.33 Mbps
Duration: 2.00 s | Packets:  11059 | Errors:      0 | Rate: 11.32 Mbps
Duration: 2.00 s | Packets:  11057 | Errors:      0 | Rate: 11.31 Mbps
Duration: 2.00 s | Packets:  11051 | Errors:      0 | Rate: 11.31 Mbps
-
Upload completed!
Duration:       10.00 s
Num packets:    55290
Num errors:     0 (retry or fail)
Rate:           11.32 Mbps
uart:~$ 
```

rlubos commented on subsys/net/lib/zperf/zperf_shell.c:
```diff
@@ -866,6 +907,23 @@ static int shell_cmd_upload(const struct shell *sh, size_t argc,
+					      "UDP does not support -i option\n");
+				return -ENOEXEC;
+			}
+			int seconds = parse_arg(&i, argc, argv);
```
We usually define variables on top of the block.

rlubos reviewed (commented):
Please check complinace results, some minor issues need fixing.

nxpadamm commented on subsys/net/lib/zperf/zperf_common.c:
```diff
@@ -189,6 +189,14 @@ int zperf_prepare_upload_sock(const struct sockaddr *peer_addr, uint8_t tos,
 		}
 	}
 
+	if (proto == IPPROTO_TCP && tcp_nodelay &&
```
The tcp_nodelay option is included in zperf params, but was not used in `zperf_prepare_upload_sock`. I assume it was intended that all zperf socket options should be possible to set in this function.

I agree it should be in separate commit, will do that.

Commit:
net: zperf: Move TCP nodelay socket option

TCP nodelay option is part of zperf upload params, but was not being
set in common zperf_prepare_upload_sock function. Move it there
to align with how other options are set.

Signed-off-by: Adam Matus <adam.matus@nxp.com>

Commit:
net: zperf: Add periodic reporting for upload

Add option for zperf tcp upload that will enable periodic result reporting.
This is useful for monitoring performance swings during a longer session.

Signed-off-by: Adam Matus <adam.matus@nxp.com>
